### ISSUE TITLE
Problem with Query Definition in Schema

### ISSUE THREAD

**Issue by dirkkolb** [kensho-technologies/graphql-compiler#274]
Your current test schema looks like this:

```
type RootSchemaQuery {
        Animal: Animal
        BirthEvent: BirthEvent
        Entity: Entity
        Event: Event
        FeedingEvent: FeedingEvent
        Food: Food
        FoodOrSpecies: FoodOrSpecies
        Location: Location
        Species: Species
        UniquelyIdentifiable: UniquelyIdentifiable
    }
```

If I understood GraphQL and the queries correctly, you should return an `[Animal]`  instead of a single `Animal`.  But this causes some problems within your compiler.

**Comment by obi1kenobi** [comment]
You're absolutely right, we generally can return multiple `Animal` objects from the top-level query. Good catch, and thank you for the report!

I think the compiler should allow both `Animal` and `[Animal]`, since the schema-writer (whether person or code) might know a certain object to be a singleton. But it definitely shouldn't break on `[Animal]`, so this is something we should fix.

We'll aim to get it in the next stable release.

**Comment by obi1kenobi** [comment]
I believe this is fixed in #284.
